### PR TITLE
feat(sink): support redis cluster url

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

https://github.com/risingwavelabs/risingwave/issues/15622

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note
we support redis cluster url.
If redis is a cluster, we need use json array , like 
redis.url= '["redis://redis-server:6379/","redis://redis-server:6380/","redis://redis-server:6381/]'

if redis is not a cluster,we need use string like (Same way as before.)
redis.url= 'redis://redis-server:6379/'

</details>

### CODE REVIEW

File: src/connector/src/sink/redis.rs [hunk 1]
```diff
@@ -44,13 +48,57 @@ pub const VALUE_FORMAT: &str = "value_format";
...
-        let client = RedisClient::open(self.url.clone())?;
-        Ok(client)
+    pub async fn build_conn(&self) -> ConnectorResult<RedisConn> {
+        if self.url.is_empty() {
+            Err(SinkError::Redis("Missing redis.url".to_string()).into())
+        } else if self.url.len() == 1 {
+            let client = RedisClient::open(self.url.get(0).unwrap().clone())?;
+            Ok(RedisConn::Single(
+                client.get_multiplexed_async_connection().await?,
+            ))
```
Comment (neverchanje) [inline src/connector/src/sink/redis.rs:97]: I wonder what if we write to a master-slave mode redis?

File: integration_tests/redis-sink/create_sink.sql [hunk 2]
```diff
@@ -3,31 +3,31 @@ FROM
     bhv_mv WITH (
     primary_key = 'user_id',
     connector = 'redis',
-    redis.url= 'redis://redis:6379/',
+    redis.url= '["redis://127.0.0.1:6379/"]',
```
Comment (wenym1) [inline integration_tests/redis-sink/create_sink.sql:6]: Should it be `redis:6379`?

File: src/connector/src/sink/redis.rs [hunk 3]
```diff
@@ -44,13 +48,57 @@ pub const VALUE_FORMAT: &str = "value_format";
 #[derive(Deserialize, Debug, Clone, WithOptions)]
 pub struct RedisCommon {
     #[serde(rename = "redis.url")]
```
Comment (wenym1) [inline src/connector/src/sink/redis.rs:50]: I think for single node setting, we should allow not wrapping the address with `[]` like `url="redis:6379"` to be compatible with the original way of config.

File: src/connector/src/sink/redis.rs [hunk 4]
```diff
@@ -44,13 +48,57 @@ pub const VALUE_FORMAT: &str = "value_format";
...
-    pub url: String,
+    pub url: Vec<String>,
+}
+pub enum RedisConn {
+    Cluster(ClusterConnection),
```
Comment (wenym1) [inline src/connector/src/sink/redis.rs:54]: What's the difference between `MultiplexedConnection` and `ClusterConnection`? Can single node connection be seen as a special case of cluster connection that has only one node? Or there is any special reason to use `MultiplexedConnection`, such as better performance.

Comment (xxhZs) [reply src/connector/src/sink/redis.rs:54]: Tried connecting to MultiplexedConnection  with single node, can't get it to run

File: src/connector/src/sink/redis.rs [hunk 5]
```diff
@@ -46,11 +49,73 @@ pub struct RedisCommon {
...
+    }
+}
 
 impl RedisCommon {
-    pub(crate) fn build_client(&self) -> ConnectorResult<RedisClient> {
-        let client = RedisClient::open(self.url.clone())?;
-        Ok(client)
+    pub async fn build_conn(&self) -> ConnectorResult<RedisConn> {
+        let url: Value =
+            serde_json::from_str(&self.url).map_err(|e| SinkError::Config(anyhow!(e)))?;
```
Comment (wenym1) [inline src/connector/src/sink/redis.rs:91]: May add more details about the anyhow error, like `anyhow!("failed to parse url config: {:?}", e.as_report())`

File: src/connector/src/sink/redis.rs [hunk 6]
```diff
@@ -46,11 +49,73 @@ pub struct RedisCommon {
     #[serde(rename = "redis.url")]
     pub url: String,
 }
+pub enum RedisConn {
```
Comment (wenym1) [inline src/connector/src/sink/redis.rs:52]: May add doc about the difference between the two types of conn.

**Review by wenym1 — APPROVED** [review]
Rest LGTM!

Need to add more details on the release note part.

**Review by hzxa21 — APPROVED** [review]
`Cargo.lock` LGTM

**Pull request merged**